### DISCUSSION TITLE
issue: the result of handleSubmit is passed to SyntheticEvent after reset

### DISCUSSION THREAD

**Discussion by shongs27** [react-hook-form/react-hook-form#11000] [Q&A] [answered]
<img width="699" alt="스크린샷 2023-10-04 오전 11 04 57" src="https://github.com/react-hook-form/react-hook-form/assets/55541745/1b1a447f-09c1-4718-8599-7bedb45a2690">
<img width="270" alt="스크린샷 2023-10-04 오전 11 11 11" src="https://github.com/react-hook-form/react-hook-form/assets/55541745/53941e0b-bccb-4f42-92dd-e119ae3f1efb">

The code was complicated and could not be attached. If it is a problem, I will attach it later.

First of all, I wonder if it happened because I didn't understand the reaction-hook-form properly.
In the screenshot, the 'data' of the props transmitted from the handleSubmit shall always be the value of the contactor that is submitted.

If the premise is correct, the problematic part occurs after executing the reset method of useForm.
After executing useForm reset, the initialValues set in defaultValues must be delivered to handleSubmit, and SyntheticEvent is being delivered.

What could be the problem? If you take a picture with getValues(), you can see the data normally.

**Comment by Moshyfawn** [comment]
Can you show how you call `reset` and `onSubmit`?

**Reply by shongs27** [reply]
I think a screenshot would be easier to see than a code attachment
SearchForm Component is SearchFormContainer

<img width="718" alt="스크린샷 2023-10-05 오전 11 01 31" src="https://github.com/react-hook-form/react-hook-form/assets/55541745/09e105ff-d5c6-4e92-96a2-e648a5f8d30e">
<img width="740" alt="스크린샷 2023-10-05 오전 11 02 39" src="https://github.com/react-hook-form/react-hook-form/assets/55541745/fcfe9b82-1a8d-4c7d-9b1a-0e10a8ce9082">

It is confirmed that the event of submit button is delivered.

**Reply by s0h311** [reply]
Hi @shongs27 you should pass the function that you want to call to `handleSubmit`. Let's say you want to save some data:

```JSX
...
const onSave(data) => {
  // save the data
}
return (
  <form onSubmit={handleSubmit(onSave)} >
    // form
  </form> 
)
```

Let me know, if this fixes your problem

**Reply by shongs27** [reply]
@s0h311 
Thanks for the suggestion
However, there is no problem in delivering the form data(the part of the prop doing console.log) I want in the beginning. The problematic point is that the form data delivered **after pressing the reset button** is changed to Synthetic Event <onClick>

**Reply by shongs27** [reply, marked as answer]
<img width="655" alt="스크린샷 2023-10-11 오전 11 46 58" src="https://github.com/react-hook-form/react-hook-form/assets/55541745/7b8edb0e-a354-43c1-98d0-7f323ea5f891">

There may be a problem with the delivery process, but I looked at it again after receiving a hint, and found an error
It seems that an error occurred because the reset button event was transmitted to the prop values in reset.
The onReset part has changed
